model/host/host.go

```diff
@@ -810,12 +810,16 @@ func (h *Host) SetProvisioning(ctx context.Context) error {
 }
 
 // SetDecommissioned sets the host as decommissioned. If checkTaskGroup is set,
-// we only update the host if it hasn't started running a task group.
-func (h *Host) SetDecommissioned(ctx context.Context, user string, checkTaskGroup bool, logs string) error {
+// we only update the host if it hasn't started running a task group. If terminateIfBusy
+// isn't set, only update the host if there is no running task.
+func (h *Host) SetDecommissioned(ctx context.Context, user string, checkTaskGroup, terminateIfBusy bool, logs string) error {
 	query := bson.M{}
 	if checkTaskGroup {
 		query[RunningTaskGroupKey] = bson.M{"$eq": ""}
 	}
+	if !terminateIfBusy {
+		query[RunningTaskKey] = bson.M{"$eq": ""}
```

Comment:
This has to check for a nonexistent field rather than empty string because [its BSON tag is omitempty](https://github.com/evergreen-ci/evergreen/blob/8f65301db2c88bc808cf828d3306ad0aa25ba957/model/host/host.go#L75) and [clearing the running task `$unset`s the field](https://github.com/evergreen-ci/evergreen/blob/8f65301db2c88bc808cf828d3306ad0aa25ba957/model/host/host.go#L1633-L1644), which deletes it from the document entirely The easiest way to do this is to look for `query[RunningTaskKey] = nil` (which checks for both null and nonexistent BSON keys).

Same with the check above for `checkTaskGroup`, I'm not sure `checkTaskGroup` would ever check it correctly because looking for the empty string is different from looking for an unset field.

Reply:
I used "$exists": false because it looks like that's what we do in other parts of our code https://github.com/bynn/evergreen/blob/dc6efb80dc3bb0d805282dfc881882a6b0d10013/model/host/db.go#L234

Reply:
That also works, SGTM!

The only difference is that key = nil checks for both `$exists: false` (no key in the doc) and key = `null` (the key is in the doc but the value is null). In this case, using `$exists: false` should be fine.